### PR TITLE
Task comparator

### LINKED ISSUE #115: Add sorting functionality for task list
Chronological. Somedays at end in any order

### PR DESCRIPTION
**Pull request by kabir-plod**
Comparator puts tasks in chronological order first, then alphabetical.
Future extension: order by priority before lexicographical.

Closes #115

### CODE REVIEW

**Comment by kabir-plod** [comment]
Please squash and merge

**Comment by KnytJC** [comment]
If we only compare the end date, the event tasks that start earlier but end later than a particular task will be put after it. What do you think?

File: src/test/java/seedu/address/model/task/TaskEndDateComparatorTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		
+		Task deadline4 = new Task(new Name("4 deadline 4pm 25-12-2016"), new TaskType("deadline"), new Status("done"), 
+				Optional.empty(), Optional.of(LocalDateTime.of(2016, 12, 25, 16, 0)), new UniqueTagList());
+		
+		Task someday5 = new Task(new Name("5 someday"), new TaskType("someday"), new Status("done"), 
+				Optional.empty(), Optional.empty(), new UniqueTagList());
+		
+		Task someday6 = new Task(new Name("6 someday"), new TaskType("someday"), new Status("done"), 
+				Optional.empty(), Optional.empty(), new UniqueTagList());
+
```
Comment (kabir-plod) [inline src/test/java/seedu/address/model/task/TaskEndDateComparatorTest.java]: Noted, will do

**Comment by kabir-plod** [comment]
Made changes suggested by @KnytJC and @kelvinnpc

**Comment by KnytJC** [comment]
reviewed

**Comment by kelvinnpc** [comment]
reviewed but ListCommandTest may give problem after merging because the order written in the test is not in chronological order

**Comment by kabir-plod** [comment]
Done and list command tests still failing? @kelvinnpc

**Pull request merged**